src/table/use-table-focus-navigation.ts

```diff
@@ -0,0 +1,166 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { RefObject, useCallback, useEffect, useMemo, useRef } from 'react';
+import { TableProps } from './interfaces';
+
+function iterateTableCells<T extends HTMLElement>(
+  table: T,
+  func: (cell: HTMLTableCellElement, rowIndex: number, columnIndex: number) => void
+) {
+  table.querySelectorAll('tr').forEach((row: HTMLTableRowElement, rowIndex: number) => {
+    row.querySelectorAll('td').forEach((cell, cellIndex) => {
+      func(cell, rowIndex, cellIndex);
+    });
+  });
+}
+
+/**
+ * This hook is used to navigate between table cells using the keyboard arrow keys.
+ * All the functionality is implemented in the hook, so the table component does not
+ * need to implement any keyboard event handlers.
+ * @param enable - Toggle functionality of the hook
+ * @param tableRoot - A ref to a table container. Ideally the root element of the table (<table>); tbody is also acceptable.
+ * @param columnDefinitions - The column definitions for the table.
+ * @param numRows - The number of rows in the table.
+ */
+function useTableFocusNavigation<T extends { editConfig?: TableProps.EditConfig<any> }>(
+  selectionType: 'single' | 'multi' | 'none' = 'none',
```

Comment:
This argument should not have an own default value, because it always must come from the table